### PR TITLE
planner: split test data from test cases

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

When there is a change that affects lots of test cases in planner, it is very diffcult to update the test cases.

### What is changed and how it works?

Split test data from test logic so we can generate test cases by one command:`go test --record`.
 This PR uses `TestIndexHint` as a sample. We can split more tests once the test framework is settled.

### Check List 

Tests 

 - No code

Code changes

 - None

Side effects

 - None

Related changes

 - None

Release note

 - None

### CODE REVIEW

File: planner/core/physical_plan_test.go [hunk 1]
```diff
@@ -34,14 +45,150 @@ var _ = Suite(&testPlanSuite{})
...
+		return res, err
+	}
+	// Remove comments, since they are not allowed in json.
+	re := regexp.MustCompile("(?s)//.*?\n")
+	err = json.Unmarshal(re.ReplaceAll(byteValue, nil), &res)
+	return res, err
+}
+
+// GetTestCases gets the test cases for a test function.
+func (t *testData) GetTestCases(c *C, in interface{}, out interface{}) {
```
Comment (winoros) [inline planner/core/physical_plan_test.go]: Can we move some of them to `util/testutil`?

Comment (lamxTyler) [reply planner/core/physical_plan_test.go]: Yes, all of the framework related can be moved to there.

File: planner/core/testdata/plan_suite_in.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[
+  {
+    "name": "TestIndexHint",
+    "cases": [
+      // simple case
```
Comment (lzmhhh123) [inline planner/core/testdata/plan_suite_in.json]: Does the `json` file support comments?

Comment (lamxTyler) [reply planner/core/testdata/plan_suite_in.json]: No, so before parse, the comment will be removed.

**Pull request merged**